src/Lean/Data/Json/FromToJson.lean

```diff
@@ -28,6 +28,9 @@ instance : ToJson Json := ⟨id⟩
 instance : FromJson JsonNumber := ⟨Json.getNum?⟩
 instance : ToJson JsonNumber := ⟨Json.num⟩
 
+instance : FromJson Empty where
+  fromJson? j := throw s!"no constructor matched JSON value '{j}'"
```

Comment:
As long as the string is a literal here, rather than something generated by `deriving`, it might save someone some debugging pain someday to write:
```suggestion
  fromJson? j := throw s!"type Empty has no constructor to match JSON value '{j}'"
```

Reply:
It might also be worth noting (in the error message) that the reason for reaching this error at all is that the JSON is ill-formed and the expected type is never meant to have JSON of this kind. That is, actually calling `fromJson?` or `toJson` on an `Empty` type at runtime is always an error.

Reply:
Thanks, will re-phrase.  I wasn't sure how detailed to be in the error but I am big fan or helpful error messages so more than happy to expand it.

Reply:
Done.